src/tensor/safetensors.rs

```diff
@@ -0,0 +1,138 @@
+use super::{CopySlice, DeviceStorage, Tensor};
+use crate::{
+    shapes::{Dtype, HasShape, Shape},
+    tensor::AsVec,
+};
+use no_std_compat::{collections::BTreeMap, path::Path, string::String, vec::Vec};
+use safetensors::tensor::{
+    serialize_to_file, Dtype as SDtype, SafeTensorError, SafeTensors, TensorView,
+};
+
+struct TensorData {
+    dtype: SDtype,
+    shape: Vec<usize>,
+    data: Vec<u8>,
+}
+
+pub struct Writer {
+    tensors: BTreeMap<String, TensorData>,
+}
+
+impl Writer {
+    pub fn new() -> Self {
+        let tensors = BTreeMap::new();
+        Self { tensors }
+    }
+
+    pub fn add<S, T, E: Dtype + SafeDtype>(mut self, key: String, tensor: Tensor<S, E>) -> Self
```

Comment:
Possible to add DeviceStorage generic to Tensor here? So we can add tensors from different devices 😄

Reply:
I did what I could, but it felt like type monkey coding. I added stuff until it stopped complaining. Not sure it's the best possible type :)